### PR TITLE
AWS needs these changes

### PR DESCRIPTION
**Pull request by weihuang-jedi**
**Description**

workflow/setup_xml.py failed on AWS, to get work, we need to modify wxflow/configuration.py.

**Type of change**

This fixes issue: https://github.com/NOAA-EMC/wxflow/issues/31

- [ x] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

This code has tested on AWS.

- [ ] pynorms
- [ ] pytests

**Checklist**

- [ x] My code follows the style guidelines of this project
- [x ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

**Comment by aerorahul** [comment]
Can you describe the failure exactly?

File: src/wxflow/configuration.py [hunk 1]
```diff
@@ -96,10 +96,9 @@ def print_config(self, files: Union[str, bytes, list]) -> None:
     def _get_script_env(cls, scripts: List) -> Dict[str, Any]:
         default_env = cls._get_shell_env([])
         and_script_env = cls._get_shell_env(scripts)
-        vars_just_in_script = set(and_script_env) - set(default_env)
-        union_env = dict(default_env)
-        union_env.update(and_script_env)
-        return dict([(v, union_env[v]) for v in vars_just_in_script])
+        for key in and_script_env.keys():
+            default_env[key] = and_script_env[key]
+        return default_env
```
Comment (aerorahul) [inline src/wxflow/configuration.py:101]: This change returns the entire env. and not just the variables set via the scripts.

**Comment by weihuang-jedi** [comment]
With current code, it returns error:

setup_xml.py failed with messages like:

Traceback (most recent call last):
File "/contrib/Wei.Huang/src/global-workflow-cloud/workflow/setup_xml.py",
line 79, in
main()
File "/contrib/Wei.Huang/src/global-workflow-cloud/workflow/setup_xml.py",
line 73, in main
xml = rocoto_xml_factory.create(f'{net}_{mode}', app_config,
rocoto_param_dict)
File
"/contrib/Wei.Huang/src/global-workflow-cloud/workflow/wxflow/factory.py",
line 71, in create
return self._builders[key](*args, **kwargs)
File
"/contrib/Wei.Huang/src/global-workflow-cloud/workflow/rocoto/gfs_forecast_only_xml.py",
line 12, in *init*
super().*init*(app_config, rocoto_config)
File
"/contrib/Wei.Huang/src/global-workflow-cloud/workflow/rocoto/workflow_xml.py",
line 28, in *init*
task_list = get_wf_tasks(app_config)
File
"/contrib/Wei.Huang/src/global-workflow-cloud/workflow/rocoto/workflow_tasks.py",
line 21, in get_wf_tasks
tasks.append(task_obj.get_task(task_name))
File
"/contrib/Wei.Huang/src/global-workflow-cloud/workflow/rocoto/tasks.py",
line 244, in get_task
return getattr(self, task_name, *args, **kwargs)()
File
"/contrib/Wei.Huang/src/global-workflow-cloud/workflow/rocoto/gfs_tasks.py",
line 29, in stage_ic
prefix = ***@***.***
***@***.***@***@***.***/atmos"
File
"/contrib/Wei.Huang/src/global-workflow-cloud/workflow/wxflow/attrdict.py",
line 84, in *missing*
raise KeyError(name)
KeyError: 'BASE_CPLIC'

Before we change tasks.py, and others, the failure are different.

**Comment by aerorahul** [comment]
The issue appears to be in the config file of the experiment and not in this code. 
Please check if the missing key is defined. 
wxflow is agnostic to HPC or cloud.

**Comment by weihuang-jedi** [comment]
OK, I'll try more at the config side.

**Comment by weihuang-jedi** [comment]
@aerorahul As you suggested, after check the code, we figured that we can avoid the changes in wxflow.
I am closing this PR now. 
Thanks,
Wei

**Pull request closed without merging**